### PR TITLE
refactor(ingestion): allow configuring ch events topic and message schema for events in the plugin server [nuke protobuf pt.2]

### PR DESCRIPTION
**Pull request by yakkomajuri**
## Problem

#9207 

## Changes

Allows us to select a few teams to produce JSON encoded event messages to a separate topic so we can test #9213 and eventually flip the switch across the board

👉 *Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review.*

### CODE REVIEW

File: plugin-server/src/config/config.ts [hunk 1]
```diff
@@ -152,6 +154,9 @@ export function getConfigHelp(): Record<keyof PluginsServerConfig, string> {
             '(advanced) maximum number of promises that a worker can have running at once in the background. currently only targets the exportEvents buffer.',
         KAFKA_PARTITIONS_CONSUMED_CONCURRENTLY:
             '(advanced) how many kafka partitions the plugin server should consume from concurrently',
+        CLICKHOUSE_DISABLE_EXTERNAL_SCHEMAS_TEAMS:
```
Comment (macobo) [inline plugin-server/src/config/config.ts:157]: Both of these vars are temporary right?

File: plugin-server/src/worker/ingestion/process-event.ts [hunk 2]
```diff
@@ -214,6 +214,14 @@ export class EventsProcessor {
         return this.pluginsServer.NEW_PERSON_PROPERTIES_UPDATE_ENABLED ?? false
     }
 
+    public clickhouseExternalSchemasEnabled(teamId: number): boolean {
+        if (this.pluginsServer.CLICKHOUSE_DISABLE_EXTERNAL_SCHEMAS) {
+            return false
+        }
+        const teams = this.pluginsServer.CLICKHOUSE_DISABLE_EXTERNAL_SCHEMAS_TEAMS.split(',').filter(String).map(Number)
```
Comment (macobo) [inline plugin-server/src/worker/ingestion/process-event.ts:221]: We shouldn't be doing this every time. Also when using inclusivity use Set data type!

```
if (!this.clickhouseExternalSchemasEnabledTeams && this.pluginsServer.CLICKHOUSE_DISABLE_EXTERNAL_SCHEMAS_TEAMS) {
  this.clickhouseExternalSchemasEnabledTeams = new Set(this.pluginsServer.CLICKHOUSE_DISABLE_EXTERNAL_SCHEMAS_TEAMS.split(',').filter(String).map(Number))
}
```

**Comment by macobo** [comment]
Not sure on the approach - could we create the new kafka topic(s) and just switch plugin server over instead of needing extra vars? Everything we do needs to handle self-hosted after all.

**Comment by yakkomajuri** [comment]
We can just switch over yes, but this is just for a sanity check to run on prod. Let's def not get it in the release.

I'll just double check everything works with a team and then switch us over

File: plugin-server/src/config/config.ts (hunk 1 above)
Comment (yakkomajuri) [reply plugin-server/src/config/config.ts:157]: yup, just for some prod testing

**Pull request merged**